### PR TITLE
[CSM] Track image tags of syscalls in activity trees

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR isolates syscalls entries in activity trees into their own `SyscallNode` struct. This is done so that we start tracking syscalls per image tags. This is a 2 step PR, next step is to update the protobuf to record these image tags.

### Motivation

Make sure syscalls are tracked like any other resources.

### CODE REVIEW

File: pkg/security/security_profile/activity_tree/process_node.go [hunk 1]
```diff
@@ -201,20 +201,31 @@ func (pn *ProcessNode) Matches(entry *model.Process, matchArgs bool, normalize b
...
+		if !dryRun {
+			pn.Syscalls = append(pn.Syscalls, NewSyscallNode(int(newSyscall), imageTag, Runtime))
+			syscallMask[int(newSyscall)] = int(newSyscall)
+			stats.SyscallNodes++
+		}
 		hasNewSyscalls = true
+		if dryRun {
+			// exit early
+			break
+		}
```
Comment (spikat) [inline pkg/security/security_profile/activity_tree/process_node.go:226]: nit: if we put the `if dryRun { break }` first, we won't need the second `if`

**Review by spikat — APPROVED** [review]
Just a small NIT, otherwise LGTM :)

**Comment by spikat** [comment]
It would be great to add some functional tests too

File: pkg/security/security_profile/activity_tree/process_node.go (hunk 1 above)
Comment (Gui774ume) [reply pkg/security/security_profile/activity_tree/process_node.go:226]: haha true 👍🏻

**Review by YoannGh — APPROVED** [review]

**Pull request merged**